File: cpp/Processor.cpp
#include <iostream>
using namespace std;

class Processor
{
    private:
        string name = "";
        int price = 0;

    public:
        Processor(){}

        Processor(string name, int price)
        {
            this->name = name;
            this->price = price;
        }

        void setName(string name)
        {
            this->name = name;
        }

        string getName()
        {
            return this->name;
        }

        void setPrice(int price)
        {
            this->price = price;
        }

        int getPrice()
        {
            return this->price;
        }

        void showProcessor()
        {
            cout << "Processor name  : " << this->getName() << endl;
            cout << "Processor Price : " << this->getPrice() << endl;
        }

        ~Processor(){}
};

File: java/Processor.java
public class Processor
{
    private String name = "";
    private int price = 0;

    Processor(){}

    Processor(String name, int price)
    {
        this.name = name;
        this.price = price;
    }

    public void setName(String name)
    {
        this.name = name;
    }

    public String getName()
    {
        return this.name;
    }

    public void setPrice(int price)
    {
        this.price = price;
    }

    public int getPrice()
    {
        return this.price;
    }

    public void showProcessor()
    {
        System.out.println("Processor name  : " + this.getName());
        System.out.println("Processor Price : " + this.getPrice());
    }
}